# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1291418485 18000
#      Fri Dec 03 18:21:25 2010 -0500
# Node ID c18526be2340de112847101ada46f57c72d86122
# Parent  3bddc4ea8141ebfd3aa9671e7079a52c5600a2e7
__go_draw_axes__.m: If not set, the unset the gnuplot {tblr}margins.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2010-12-03  Ben Abbott  <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m: If not set, then unset the gnuplot
+	{tblr}margins. Needed when using subplots with differing
+	activepositionproperty.
+
 2010-12-03  Konstantinos Poulios <logari81@gmail.com>
 
 	* plot/pbaspect.m, plot/daspect.m: Fix handling of input array of
 	length 2.
 
 2010-12-02  Ben Abbott <bpabbott@mac.com>
 
 	* image/image.m: Only flip y-axis if nextplot property equal
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -112,16 +112,20 @@ function __go_draw_axes__ (h, plot_strea
         if (strcmpi (axis_obj.dataaspectratiomode, "manual"))
           sz_str = sprintf ("set size ratio %.15g", -dr);
         else
           sz_str = "set size noratio";
         endif
         sz_str = sprintf ("%s %.15g, %.15g;\n", sz_str, pos(3), pos(4));
       endif
     else ## activepositionproperty == outerposition
+      fprintf (plot_stream, "unset tmargin;\n")
+      fprintf (plot_stream, "unset bmargin;\n")
+      fprintf (plot_stream, "unset lmargin;\n")
+      fprintf (plot_stream, "unset rmargin;\n")
       fprintf (plot_stream, "set origin %g, %g;\n", pos(1:2))
       sz_str = "";
       if (strcmpi (axis_obj.dataaspectratiomode, "manual"))
         sz_str = sprintf ("ratio %g", -dr);
       else
         sz_str = "noratio";
       endif
       sz_str = sprintf ("set size %s %g, %g;\n", sz_str, pos(3:4));
